Sources/Swift/Helper/DataDeserialization.swift

```diff
@@ -29,4 +31,154 @@
         return nil
       }
     }
+    
+    //swiftlint:disable cyclomatic_complexity function_body_length
+    @objc(envelopeWithData:) public static func envelope(with data: Data) -> SentryEnvelope? {
+        var envelopeHeader: SentryEnvelopeHeader?
+        let bytes = [UInt8](data)
+        var envelopeHeaderIndex: Int = 0
+        
+        for i in 0..<data.count {
+            if bytes[i] == UInt8(ascii: "\n") {
+                envelopeHeaderIndex = i
+                // Envelope header end
+                let headerData = data.subdata(in: 0..<i)
+                #if DEBUG
+                  if let headerString = String(data: headerData, encoding: .utf8) {
+                    SentrySDKLog.debug("Header \(headerString)")
+                  }
+                #endif
+                do {
+                    let headerDictionary = try JSONSerialization.jsonObject(with: headerData) as? [String: Any]
```

Comment:
`m`: If I'm not mistaken, we can move the catch block up here, cause nothing below uses try. This would be better for readability, because the logging of the error is closer to the try here.